src/hooks/kratos/useWhoami.ts

```diff
@@ -1,39 +1,41 @@
 import { Session } from '@ory/kratos-client';
 import { useEffect, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
 import { useKratosClient } from './useKratosClient';
 
 export const useWhoami = () => {
+  const { t } = useTranslation();
   const [session, setSession] = useState<Session>();
   const [error, setError] = useState<string>();
   const [loading, setLoading] = useState<boolean>(true);
   const [isAuthenticated, setIsAuthenticated] = useState<boolean>(false);
-  const DEFAULT_ERROR_MESSAGE = "Can't get session information!";
   const kratosClient = useKratosClient();
 
   useEffect(() => {
-    kratosClient
-      .toSession()
-      .then(result => {
-        if (result.status === 200) {
-          setSession(result.data);
-          setIsAuthenticated(true);
-        } else if (result.status === 401) {
+    if (kratosClient)
+      kratosClient
+        .toSession()
+        .then(result => {
+          if (result.status === 200) {
```

Comment:
The below code can be generalized as `setIsAuthenticated(result.status === 200)` and then handle the different cases if neccessary.